plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/OTelTelemetryPlugin.java

```diff
@@ -58,20 +60,39 @@ public List<Setting<?>> getSettings() {
 
     @Override
     public Optional<Telemetry> getTelemetry(TelemetrySettings telemetrySettings) {
-        return Optional.of(telemetry(telemetrySettings));
+        return telemetry(telemetrySettings);
     }
 
     @Override
     public String getName() {
         return OTEL_TRACER_NAME;
     }
 
-    private Telemetry telemetry(TelemetrySettings telemetrySettings) {
-        final OpenTelemetrySdk openTelemetry = OTelResourceProvider.get(telemetrySettings, settings);
-        return new OTelTelemetry(
-            new OTelTracingTelemetry<>(openTelemetry, openTelemetry.getSdkTracerProvider()),
-            new OTelMetricsTelemetry<>(openTelemetry.getSdkMeterProvider())
-        );
+    private Optional<Telemetry> telemetry(TelemetrySettings telemetrySettings) {
+        final Optional<OpenTelemetrySdk> openTelemetry = OTelResourceProvider.get(telemetrySettings, settings);
+        if (openTelemetry.isPresent()) {
+            return Optional.of(
+                new OTelTelemetry(getTracingTelemetry(settings, openTelemetry.get()), getMetricsTelemetry(settings, openTelemetry.get()))
+            );
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private MetricsTelemetry getMetricsTelemetry(Settings settings, OpenTelemetrySdk openTelemetry) {
+        if (TelemetrySettings.METRICS_FEATURE_ENABLED_SETTING.get(settings) == true) {
```

Comment:
Here is the deal, those settings are declared as clusters ones which means that could be adjusted dynamically to a whole cluster. This is not going to work with the plugin if implemented this way.

Reply:
These are static settings only both the dynamic cluster settings.

Reply:
Oh missed that (sorry) part .... because we have yet another one, which becomes confusing now:

```
public static final Setting<Boolean> TRACER_ENABLED_SETTING = Setting.boolSetting(
        "telemetry.tracer.enabled",
        false,
        Setting.Property.NodeScope,
        Setting.Property.Dynamic
    );
```

This has to go away I believe: we have too many knobs to control  the same behaviour. I think if we keep only the ones you have introduced in this pull request, that would be an option.

The other thing we have talked about is dynamic on / off, and we had complications for metrics there.

Reply:
Another thing to consider: these settings are specific to the plugin, there is no reason to make them available to everyone: plugin implementors have to know about them, otherwise they will be ignored and have zero effect. 

In this regards, `TRACER_ENABLED_SETTING` is taking a different stance: it fences the tracing in core from tracing provider (plugin). I don't think we should have this XXX_FEATURE_ENABLED_XXX settings, it would be better to find out how we could have   `METRICS_ENABLED_SETTING` in such a way that a) it won't produce weird results b) but at the same time fences core from metrics provider  (plugin).

Reply:
> Alternatively, we could keep these settings (and drop TRACER_ENABLED_SETTING) but implementation only reside in core - disable tracing / metrics completely on initialization, and never ask any plugins for it.

I got you, and actually it make sense. Only issue is the way OpenTelemetry starts here we need to initialise both Metrics and Tracing but will be fenced from the core based on the flag. Some extra tracer/metrics threads would be there but I think that should be ok.  

TRACER_ENABLED_SETTING is also a good knob to disable the traces on running cluster in case some resource pressure, enable for on-demand tracing or debugging. Metrics are little different here as they will occupy mostly the same resources irrespective of whether counters are updated or not.

Reply:
@reta, I have made the changes. Please take a look as per your convenience. Thanks!